File: document/cgi/show/file_list.pl
#!/usr/bin/perl

use strict;
use warnings;
use Cwd 'abs_path';
use File::Basename;
# require '/Users/tayou/Desktop/tayou/42_webserv/document/cgi/show/file_list_util.pl';
require '/Users/jinhyeop/Desktop/project/webserv/document/cgi/show/file_list_util.pl';
# require '/Users/yejlee2/42cursus/webserv/document/cgi/show/file_list_util.pl';

print "Content-Type: text/html\r\n\r\n";

# my $uploadDir = '/Users/tayou/Desktop/tayou/42_webserv/document/uploaded';
my $uploadDir = '/Users/jinhyeop/Desktop/project/webserv/document/uploaded';
# my $uploadDir = '/Users/yejlee2/42cursus/webserv/document/uploaded';
opendir (my $dh, $uploadDir) or die "Cannot open directory: $!";
my @files = grep { !/^\./ && -f "$uploadDir/$_" } readdir($dh);
closedir($dh);

my $htmlStringList = getHTMLStringList(@files);

# open (my $htmlFile, '<', '/Users/tayou/Desktop/tayou/42_webserv/document/html/file_list.html') or die "Cannot open HTML template: $!";
open (my $htmlFile, '<', '/Users/jinhyeop/Desktop/project/webserv/document/html/file_list.html') or die "Cannot open HTML template: $!";
# open (my $htmlFile, '<', '/Users/yejlee2/42cursus/webserv/document/html/file_list.html') or die "Cannot open HTML template: $!";
my $htmlContent = do { local $/; <$htmlFile> };
$htmlContent =~ s/{fileList}/$htmlStringList/g;

print $htmlContent;

File: document/cgi/show/file_list.py
#!/usr/bin/python3

import sys, os
sys.path.append(os.path.dirname(os.path.abspath(os.path.dirname(__file__))))
from util import util
import cgitb, file_list_util

cgitb.enable()
uploadDir = '/Users/jinhyeop/Desktop/project/webserv/document/uploaded'
# uploadDir = '/Users/tayou/Desktop/tayou/42_webserv/document/uploaded'
# uploadDir = '/Users/yejlee2/42cursus/webserv/document/uploaded'
fileList = util.getFileList(uploadDir)
htmlStringList = file_list_util.getHTMLStringList(fileList)
print("Content-Type: text/html\r\n\r\n", end="")
htmlFile = open('/Users/jinhyeop/Desktop/project/webserv/document/html/file_list.html', 'r')
# htmlFile = open('/Users/tayou/Desktop/tayou/42_webserv/document/html/file_list.html', 'r')
# htmlFile = open('/Users/yejlee2/42cursus/webserv/document/html/file_list.html', 'r')
htmlContent = htmlFile.read()
htmlContent = htmlContent.format(fileList = htmlStringList)
print(htmlContent)
